proxy/src/connection.rs

```diff
@@ -68,16 +78,46 @@ impl BoundPort {
     // This ensures that every incoming connection has the correct options set.
     // In the future it will also ensure that the connection is upgraded with
     // TLS when needed.
-    pub fn listen_and_fold<T, F, Fut>(self, executor: &Handle, initial: T, f: F)
+    pub fn listen_and_fold<T, F, Fut>(self, handle: &Handle, initial: T, f: F)
+        -> Box<Future<Item = (), Error = io::Error> + Send + 'static>
```

Comment:
In 2 days, this can just return `impl Future`, and the `Send` bounds can be propagated by the generic arguments!